### PR TITLE
chore(madsim,deps): use patched `sqlx` managed by the `madsim-rs` organization instead of a personal repository

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We use patched sqlx from the madsim fork, instead of the one in my personal fork of sqlx.

Please review the changes in that fork prior to reviewing this PR.

This accomplishes two things:
1. Let the Madsim Org manage the fork. We can have PR reviews each time we rebase the patch.
2. Inside the fork will adopt a rebase workflow, and continually patch for madsim on top of the tagged commit corresponding to the latest sqlx version.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by yezizp2012 — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -9747,7 +9747,7 @@ dependencies = [
  "indoc",
  "libc",
  "memoffset",
- "parking_lot 0.12.1",
+ "parking_lot 0.11.2",
```
Comment (hzxa21) [inline Cargo.lock:9750]: Is the downgrade here expected?

**Review by fuyufjh — APPROVED** [review]

File: Cargo.lock (hunk 1 above)
Comment (kwannoel) [reply Cargo.lock]: Reverted it. Seems really strange. Each time I run `cargo update sqlx`, it keeps updating this dependency as well, for some reason.

**Comment by BugenZhao** [comment]
> We use patched sqlx from the risingwave fork: [risingwavelabs/sqlx#1](https://github.com/risingwavelabs/sqlx/pull/1), instead of the one in my personal fork of sqlx.

As the changes are solely to make `sqlx` compatible with `madsim`, I slightly prefer keeping it maintained under `madsim-rs`, especially when considering that we all have write permissions to `madsim-rs` now.

Otherwise, should we also move other folks, like `tonic`, `rdkafka`, `tokio-postgres` or even `tokio-stream` into `risingwavelabs` org?

**Comment by kwannoel** [comment]
> > We use patched sqlx from the risingwave fork: [risingwavelabs/sqlx#1](https://github.com/risingwavelabs/sqlx/pull/1), instead of the one in my personal fork of sqlx.
> 
> As the changes are solely to make `sqlx` compatible with `madsim`, I slightly prefer keeping it maintained under `madsim-rs`, especially when considering that we all have write permissions to `madsim-rs` now.
> 
> Otherwise, should we also move other forks, like `tonic`, `rdkafka`, `tokio-postgres` or even `tokio-stream` into `risingwavelabs` org?

Done. https://github.com/madsim-rs/sqlx/pull/3.

File: Cargo.toml [hunk 2]
```diff
@@ -350,10 +350,11 @@ getrandom = { git = "https://github.com/madsim-rs/getrandom.git", rev = "e79a7ae
...
 tokio-retry = { git = "https://github.com/madsim-rs/rust-tokio-retry.git", rev = "95e2fd3" }
 tokio-postgres = { git = "https://github.com/madsim-rs/rust-postgres.git", rev = "ac00d88" }
-# NOTE(kwannoel): patch sqlx to use a version compatible with madsim.
-# If we can merge in: https://github.com/madsim-rs/sqlx/pull/2,
-# we can change it to patch madsim version instead.
-sqlx = { git = "https://github.com/kwannoel/sqlx.git", rev = "ddf222f56cf99f865231a5383053645c6ea05ba3" }
+
+# sqlx version: v0.7.4
+# patch diffs: https://github.com/madsim-rs/sqlx/pull/3
+sqlx = { git = "https://github.com/risingwavelabs/sqlx.git", rev = "ff6d6d2dc0e9e8e47282fd29be006eed7ae3421a" }
```
Comment (kwannoel) [inline Cargo.toml:356]: I pushed it to `madsim`, but didn't update it here 🤦 . Regardless they are using the same commit.

I will update this later when bumping sqlx to 0.8.2: https://github.com/risingwavelabs/risingwave/pull/19145

**Pull request merged**